src/controllers/profileController.js

```diff
@@ -0,0 +1,104 @@
+
+const { Profiles, Departments, Users } = models;
+/**
+ * It contains the controller methods that handle requests to the profile
+ * endpoint entity
+ * @class profileController
+*/
+export default class profileController {
+/**
+   * It receives the social provider to use for authentication and returns an express
+   * controller function to handle the callback request that happens after an
+   * authentication attempt has happened
+   * @function callback
+   * @param {Object} req The express request object
+   * @param {Object} res The express response object
+   * @returns {Object} UserProfile The current user profile
+*/
+  static async getProfile(req, res) {
+    const { params: { userId } } = req;
+    try {
+      const { dataValues: data } = await Profiles.findOne({
+        where: { userId },
+        include: [
+          {
+            model: Departments,
+            as: 'theDepartment',
+            attributes: [['name', 'department']]
+          },
+          {
+            model: Users,
+            as: 'theUser',
+            attributes: ['firstName', 'lastName', 'email', 'isVerified']
+          }
+        ]
+      });
+      return successResponse(res, 200, 'Profile fetched successfully', data);
+    } catch (error) {
+      return errorResponse(res, 500, 'Operation not successful. Please try again later.');
+    }
+  }
+  
+  /**
+     * It receives the social provider to use for authentication and returns
+     * an express controller function that handles the initial request that
+     * seeks authentication from the social platforms
+     * @function authenticate
+     * @param {Object} req The express request object
+     * @param {Object} res The express response object
+     * @returns {Object} UserProfile The new profile just created
+  */
+  static async addProfile(req, res) {
+    try {
+      const {
+        body: {
+          department: name, phoneNumber, role, jobDescription, imageUrl
+        }
+      } = req;
+      const { userId } = req;
+      const { dataValues: { id: departmentId } } = await Departments.findOne({ where: { name } });
+      const { dataValues: data } = await Profiles.create({
+        userId, phoneNumber, departmentId, role, jobDescription, imageUrl
+      });
+      return successResponse(res, 201, 'Profile created successfully', data);
+    } catch (error) {
+      return errorResponse(res, 500, 'Operation not successful. Please try again later.');
+    }
+  }
+  
+  /**
+   * It receives the social provider to use for authentication and returns
+   * an express controller function that handles the initial request that
+   * seeks authentication from the social platforms
+   * @function authenticate
+   * @param {Object} req The social platform to authenticate with
+   * @param {Object} res The scope of data to request from the platform
+   * @returns {function} authenticatedFunction
+  */
+  static async updateProfile(req, res) {
```

Comment:
I feel we should only have an `updateProfile` wherein the user updates remaining information from necessary to be able to successfully create a trip request. I feel the `addProfile` is not neccessary.

Reply:
The reason for having this, is because if you remember, our registration endpoint only requires `email` and `password`. We're trying to make it as easy and as simple as possible to quickly sign up. So, the flow is, it is `after` signup, that the user gets to go to the page where they create their profile. The registration and profile creation are currently two different tasks the way we designed the application.